src/target/stm32h7.c

```diff
@@ -257,9 +257,20 @@ void stm32h7_configure_wdts(target_s *const target)
 bool stm32h7_probe(target_s *target)
 {
 	const adiv5_access_port_s *const ap = cortex_ap(target);
+	DEBUG_TARGET("%s: AP partno %X, part_id %X\n", __func__, ap->partno, target->part_id);
+
 	/* Use the partno from the AP always to handle the difference between JTAG and SWD */
 	if (ap->partno != ID_STM32H72x && ap->partno != ID_STM32H74x && ap->partno != ID_STM32H7Bx)
 		return false;
+
+	/* By now it's likely this is ST silicon, either H7 or MP15x_CM4 */
+	const uint32_t idcode = target_mem32_read32(target, DBGMCU_IDCODE);
+	const uint16_t dev_id = idcode & STM32H7_DBGMCU_IDCODE_DEV_MASK;
+	DEBUG_TARGET("%s: looking at device ID 0x%3X at 0x%08X\n", __func__, dev_id, DBGMCU_IDCODE);
```

Comment:
This reads like it's debugging?

Reply:
Yes, it's not essential to driver operation, but... it was essential to telling the chips apart and coding the mis-attach guardrails appropriately. I can post the logs this produced. It should not take up flash space in release firmware, but get optimized out.

Reply:
We understand that it won't impact binary size, we're mostly thinking about how this'd look in the logs and how useful it could be (or not be) - we appreciate the usefulness to your figuring out this patch, but we're trying to weigh up the future usefulness of it. After all, it's not something a typical user of even BMDA is going to want to see in the logs, but rather only someone trying to dig into part identification things, noting that the target read gets logged at the `DEBUG_PROTO` level, allowing the value to be read in that case albeit at a slightly higher amount of effort.

We think that if this is going to be kept because you feel it does have utility, then please do two small things for code base consistency: use lower-case `x` for the device ID along with making sure it's suffixed with 0 as in `%03x`, and use PRIx32 for the ID code so its printing is better guaranteed to work as in `%08" PRIx32 "`. The rest of the code base uses lower-case hex for formatted values and literals for debug, so please let's not introduce a competing different way of doing that with this.

Reply:
In my defense, `stm32l4_probe()` does a `DEBUG_INFO("ID Code: %08" PRIx32 "\n", device_id)` for DPv2/AP partno or `DBGMCU_IDCODE` of STM32L55. Other drivers don't. I'd like to keep it in some form, possibly rewording the logged string.

Reply:
This is actually a print we intend on eliminating as it's both in the wrong debug level, and quite distracting in the logs due to it printing for a fair few ST devices that aren't actually handled by the L4 support, resulting in confusing and unhelpful logs with spurious extra lines. That's part of why we're concerned about continuing utility, because that log line from 8845a222 / a6a8606e does not have utility and introduces a spurious source of confusion in the logs